apps/site/assets/ts/schedule/components/schedule-finder/ServiceOptGroup.tsx

```diff
@@ -1,34 +1,67 @@
 import React, { ReactElement } from "react";
-import ServiceOption from "./ServiceOption";
-import {
-  ServiceOptGroupName,
-  ServiceByOptGroup
-} from "../../../helpers/service";
+import { serviceDays, startToEnd } from "../../../helpers/service";
+import { Service } from "../../../__v3api";
+import { shortDate } from "../../../helpers/date";
 
 interface Props {
-  group: ServiceOptGroupName;
   label: string;
-  services: ServiceByOptGroup[];
+  services: Service[];
   multipleWeekdays: boolean;
 }
 
 const ServiceOptGroup = ({
-  group,
   label,
   services,
   multipleWeekdays
 }: Props): ReactElement<HTMLElement> | null =>
   services.length === 0 ? null : (
-    <optgroup key={group} label={label}>
-      {services.map(service => (
-        <ServiceOption
-          key={service.service.id + service.servicePeriod}
-          service={service.service}
-          group={group}
-          servicePeriod={service.servicePeriod}
-          multipleWeekdays={multipleWeekdays}
-        />
-      ))}
+    <optgroup label={label}>
+      {services.map(service => {
+        const isMultipleWeekday =
+          multipleWeekdays &&
+          service.type === "weekday" &&
+          service.typicality !== "holiday_service";
+
+        const startDate = new Date(service.start_date);
+        const endDate = new Date(service.end_date);
+        let optionText = "";
+
+        if (service.typicality === "unplanned_disruption") {
+          const stormServicePeriod =
+            startDate === endDate
+              ? shortDate(startDate)
+              : startToEnd(startDate, endDate);
+
+          optionText = `Storm service, ${stormServicePeriod}`;
+        } else if (service.typicality === "holiday_service") {
+          optionText = service.added_dates
+            .map(
+              addedDate =>
+                `${service.added_dates_notes[addedDate]}, ${shortDate(
```

Comment:
Do we know if `added_dates_notes` will always be there for a holiday service? I seem to recall a recent conversation about how it wasn't present for some holiday.

Reply:
@phildarnowsky Do you know?

Reply:
@digitalcora @thecristen as I understood what we agreed on with the data team, `added_dates_notes` should be present for all holidays. Is that how you remember it @paulswartz @handorff ?

Reply:
@digitalcora I have updated this bit of logic, so that it'll fall back to a different bit of text if for whatever reason there is no `added_dates_notes` present.